### PR TITLE
Update setup

### PR DESCRIPTION
**Pull request by ateucher**
Replaces closed #7.

Remaining tasks from #7:

- [x] Check documentation of `create_raster_template()`
- [x] Confirm S3 approach for `create_base_vectors()` (https://github.com/ninoxconsulting/PEMprepr/pull/7#discussion_r1771980682)
- [x] in `get_*()` functions, action when `nrow() == 0` (https://github.com/ninoxconsulting/PEMprepr/pull/7#discussion_r1772014853)
- [x] Review `create_base_vectors()` for efficiency
- [x] Start tests for `create_template_raster()`
- [ ] Tests for `get_cded_dem()`
- [ ] What's going on with `create_bgc_template()`?

### CODE REVIEW

**Comment by ateucher** [comment]
@gcperk I think this is back to you now!

Can you:

1. See if you're happy with the S3 approach to `create_base_vectors()`?
2. Make a consistent approach to the carious `get_*()` functions when the output has 0 rows? I think throwing a warning, returning NULL, and not writing anything is the right approach, as you've done in `get_transmission_lines()`. Might be able to create a helper function to use in all of them...
3. `get_cded_dem()` could use some tests. You can copy my approach for `create_base_vectors()` and `create_temple_raster()` - let me know if it's not clear.
4. Decided if implementing `create_bgc_template()` is in scope for this PR, or come in another one? I'm inclined to say let's do it in a separate PR

Thanks!!

**Comment by gcperk** [comment]
Hey @ateucher , 
The s3 methods look good, although I'm pretty new to writing those so will be happy to go with your recommendations. I also added the missing if/else switch for the create_base_vector(). I think I prefer to draw a line here and get this branch merged before your comments on 3 and 4.

File: R/create_base_vectors.R [hunk 1]
```diff
@@ -1,299 +1,477 @@
...
+
+    # deciduous predominant polygons
+
+    vri_decid <- vri |>
+      dplyr::filter(.data$SPECIES_CD_1 %in% c("AT", "EP")) # note might need to adjust for some areas of interest
+
+    if (nrow(vri_decid) > 0) {
+      sf::st_write(vri_decid, fs::path(out_dir, "vri_decid.gpkg"), append = FALSE)
+      cli::cat_line()
+      cli::cli_alert_success("VRI class 1 to 3 (0-60yrs) polygons subset and to written to {.path {out_dir}}")
```
Comment (ateucher) [inline R/create_base_vectors.R:181]:
```suggestion
      cli::cli_alert_success("deciduous polygons subset and to written to {.path {out_dir}}")
```

Just a missed copy-paste edit I think?

**Review by ateucher — COMMENTED** [review]
I think this looks good Gen!

**Comment by gcperk** [comment]
Ticked the last two items on this PR so deleting this branch to keep everything tidy

**Pull request merged**